packages/site/src/content/docs/rules/node/unsupportedNodeAPIs.mdx

```diff
@@ -0,0 +1,63 @@
+---
+description: "Ensure Node.js built-in APIs are supported in the target Node.js version."
+title: "unsupportedNodeAPIs"
+topic: "rules"
+---
+
+import { Tabs, TabItem } from "@astrojs/starlight/components";
+
+import { RuleEquivalents } from "~/components/RuleEquivalents";
+import RuleSummary from "~/components/RuleSummary.astro";
+
+<RuleSummary plugin="node" rule="unsupportedNodeAPIs" />
+
+Using Node.js APIs that are not supported in your target Node.js version can cause runtime errors.
+This rule detects usage of Node.js built-in APIs that were introduced in newer versions and may not be available in older environments.
+
+## Examples
+
+<Tabs>
+<TabItem label="❌ Incorrect">
+
+```ts
+import { webcrypto } from "crypto";
+
+import * as fs from "fs";
+const blob = await fs.openAsBlob("file.txt");
+
+import { Readable } from "stream";
+const webStream = Readable.toWeb(nodeStream);
+```
+
+</TabItem>
+<TabItem label="✅ Correct">
+
+```ts
+import { randomBytes } from "crypto";
+
+import { readFile } from "fs";
+const content = await readFile("file.txt");
+
+import { Readable } from "stream";
+const nodeStream = new Readable();
+```
+
+</TabItem>
+</Tabs>
+
+## Options
+
+This rule is not configurable.
+
+## When Not To Use It
+
+If your codebase has a well-defined minimum Node.js version requirement that supports all the APIs you're using, or if you're confident that all environments where your code runs support the necessary Node.js version, you might choose not to enable this rule.
```

Comment:
Just because "you're confident that all environments where your code runs support the necessary Node.js version" doesn't mean it'll stay that way, that's the point of this rule. Perhaps a rational more like this makes more sense?

```suggestion
If you're using environments other than Node.js, or if you're confident you or your bundling process is polyfilling the desired features, you might choose not to enable this rule.
```

I think the main time this makes sense is if you're a) targeting Deno/Bun/Cloudflare or b) bundling/polyfilling.

Or something along those lines but more in line with Flint's voice.